src/nprintml/label/aggregator/pcap.py

```diff
@@ -42,35 +42,31 @@ def __call__(self, compress=False, sample_size=1):
             print('  compressed nPrint shape:', npts.shape)
 
         print('Attaching labels to nPrints')
-
         labels = self.load_label(self.label_csv)
-        labels = self.rewrite_labels(labels, npts)
 
         (npt_df, missing_labels, ogns, nns) = self.attach_label(npts, labels)
 
         print('  labels attached: missing labels for:', missing_labels)
         print('    missing labels caused samples to be dropped:', (ogns - nns))
 
         return npt_df
-    
-    def rewrite_labels(self, labels, npts):
-        """Transfer original PCAP labels to new nPrint files so that labels
-        can be correctly aggregated
 
-        """
-        new_labels = []
-        stems = {}
-        # Get stems for orignal files
-        for row in labels.itertuples():
-            stems[pathlib.PurePath(row.Index).stem] = row.Label
-    
-        # Associate old labels with new nPrint files
-        for row in npts.itertuples():
-            stem = pathlib.PurePath(row.Index).stem
-            if stem in stems:
-                new_labels.append((row.Index, stems[stem]))
-
-        return pd.DataFrame(new_labels, columns=['item', 'label']).set_index('item')
+    def load_label(self, labels_csv):
+        labels = super().load_label(labels_csv)
+
+        # Enable join on file path index with .npt-derived data --
+        # strip any file suffix like .pcap (or .npt)
+        #
+        # (Unlike with former, which strictly loaded .pcap files, mapped these to .npt files, and
+        # constructed its index from actual file paths, here we are loading file path values from
+        # user-specified CSV data. As such, we won't assume too much about what suffix the user
+        # supplied -- if any -- though .pcap is suggested.)
+        #
+        # Use regular expression rather than path method in case no extension specified, etc.,
+        # (as intent is to strip it after all -- extension not meaningful):
+        labels.index = labels.index.str.replace(r'\.(pcap|npt)$', '', case=False)
```

Comment:
This seems like the easiest/simplest way to match the indices &ndash; just discard any .pcap (or .npt) file extension.

In fact, we could leave the label index as it is, and apply the file extension `.pcap` to the paths in the index of the npt dataframe; but, this way, rather than "lie" about that, we just discard this meaningless information.